README.md

```diff
@@ -127,6 +127,64 @@ client = Redis.new(semian: {
 })
 ```
 
+#### Warning: Thread unsafe
+
+Semian is currently known to be thread-unsafe, and is intended to be used by *separate processes*.
+
+Internally, semian uses ```SEM_UNDO``` for several sysv semaphore operations:
+
+* Acquire
+* Worker registration
+* Semaphore metadata state lock
+
+The intention behind SEM_UNDO is that a semaphore operation is automatically undone when the process exits.
+
+If, however, a thread performs a semop, the SEM_UNDO is on its parent process. This means that the operation
+*will not* be undone when the thread exits. This can result in the following unfavorable behavior when using
+threads:
+
+* Threads acquire a resource, but are killed and the resource ticket is never released. For a process, the
+ticket would be released by SEM_UNDO, but since it's a thread there is the potential for ticket starvation.
+This can result in deadlock on the resource.
+* Threads that register workers on a resource but are killed and never unregistered. For a process, the worker
+count would be automatically decremented by SEM_UNDO, but for threads the worker count will continue to increment,
+only being undone when the parent process dies. This can cause the number of tickets to dramatically exceed the quota.
+* If a thread acquires the semaphore metadata lock and dies before releasing it, semian will deadlock on anything
+attempting to acquire the metadata lock until the thread's parent process exits. This can prevent the ticket count
+from being updated.
+
+Moreover, a strategy that utilizes SEM_UNDO is not compatible with a strategy that attempts to the semaphores tickets manually.
+In order to support threads, operations that currently use SEM_UNDO would need to use no semaphore flag, and the calling process
+will be responsible for ensuring that threads are appropriately cleaned up.
```

Comment:
I'd add a note on supporting threads through an in-memory semaphore, or, that it's possibly by creating another API—and that we'll welcome PRs.